docs/examples/DocsExamples/Streams/StreamTcpDocTests.cs

```diff
@@ -146,6 +115,52 @@ public void Simple_server_must_initial_server_banner_echo_server()
                 connection.HandleWith(serverLogic, Materializer);
             }, Materializer);
             #endregion
+
+            var input = new AtomicReference<List<string>>(new List<string> { "Hello world", "What a lovely day", null });
+
+            string ReadLine(string prompt)
+            {
+                // TODO: implement it
```

Comment:
Have no idea how to convert it to C#
```scala
val input = new AtomicReference("Hello world" :: "What a lovely day" :: Nil)
def readLine(prompt: String): String = {
    input.get() match {
    case all @ cmd :: tail if input.compareAndSet(all, tail) => cmd
    case _ => "q"
    }
}
```

Reply:
This is a very convoluted way to having concurrent queue with 2 elements in it ("Hello world", "What a lovely day") and picking those elements in thread safe fashion. Once a `readLine` hits an empty queue a `"q"` string is returned instead of normal content.

Reply:
Thanks, It works now